### PR TITLE
feat(webhook): support custom executable credentials and additional volume mounts

### PR DESCRIPTION
**Pull request by yangspirit**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test

/kind feature

> /kind flake

**What this PR does / why we need it**:
This PR introduces webhook enhancements to support custom identity providers and secure proxy integrations. It enables external authentication mechanisms (AIP-4117) and gives users fine-grained control over how volumes and binaries are mounted and ordered in the GCS Fuse sidecar.
Specifically, this PR introduces and validates two key features:
1. **Executable-Sourced Credentials**:
   - Extends the `gke-gcsfuse/workload-identity-credential-configmap` annotation flow to support `executable`-sourced credentials.
   - Injects `GOOGLE_EXTERNAL_ACCOUNT_ALLOW_EXECUTABLES="1"` and custom environment variables into the GCS Fuse sidecar.
   - Bypasses the Projected Service Account Token volume injection since authentication is handled dynamically by the user-provided executable.
   - **HostNetwork Support**: Refined the webhook's validation check to allow `hostNetwork: true` pods to use the `executable`-sourced OIDC flow (since it does not query the local metadata server and avoids network interception conflicts), while maintaining the protective block on standard `file`-based Workload Identity.
2. **Additional Volume Mounts (`gke-gcsfuse/additional-volume-mounts`)**:
   - Allows users to mount custom volumes (such as statically linked token copier binaries and mTLS client certificates) directly into the distroless GCS Fuse sidecar container.
   - Robustly handles validation warnings for missing volumes and ignores invalid formatting.
### Verification & Test Coverage:
- Added `TestParseCredentialConfigurationConfigMap` and `TestAppendWorkloadCredentialConfigurationVolumes` unit tests in `pkg/webhook/oidc_test.go` for executable credential parsing.
- Added complete Mutating Webhook integration tests in `pkg/webhook/mutatingwebhook_test.go` covering:
  - Formatting edge cases and missing volumes for `gke-gcsfuse/additional-volume-mounts`.
  - Seamless injection of executable workload identity credentials.
  - Bypassing hostNetwork checks for executable credentials on `hostNetwork: true` pods.

**Which issue(s) this PR fixes**:

Fixes # b/495760030

**Special notes for your reviewer**:
Please verify the deferred `hostNetwork` check logic in `pkg/webhook/mutatingwebhook.go` around lines 144-158. It now correctly parses the ConfigMap first to distinguish between file-based (blocked on hostNetwork) and executable-based (allowed on hostNetwork) configurations.

**Does this PR introduce a user-facing change?**:

```release-note
feat(webhook): introduced support for executable-sourced credentials (AIP-4117) in GCS Fuse sidecar injection.
feat(webhook): added the `gke-gcsfuse/additional-volume-mounts` annotation to mount custom volumes and certificates into the sidecar.
fix(webhook): allowed executable-sourced credentials on hostNetwork pods.
```

### CODE REVIEW

File: pkg/webhook/mutatingwebhook.go [hunk 1]
```diff
@@ -136,29 +140,78 @@ func (si *SidecarInjector) Handle(ctx context.Context, req admission.Request) ad
...
+			if mount == "" {
+				continue
+			}
+			parts := strings.Split(mount, ":")
+			if len(parts) == 2 {
+				volName := strings.TrimSpace(parts[0])
+				mountPath := strings.TrimSpace(parts[1])
+
+				if volName == "" || mountPath == "" {
+					klog.Warningf("Invalid empty volume name or mount path in %s: %s", AdditionalVolumeMountsAnnotation, mount)
```
Comment (amacaskill) [inline pkg/webhook/mutatingwebhook.go]: Why do we do have warnings instead of failing? Where do we fail later by allowing invalid options

File: pkg/webhook/mutatingwebhook_test.go
Comment (amacaskill) [inline pkg/webhook/mutatingwebhook_test.go:1]: Can we add a unit test for mount items containing more than 1 colon or no colons at all?

File: pkg/webhook/mutatingwebhook.go [hunk 2]
```diff
@@ -136,29 +140,78 @@ func (si *SidecarInjector) Handle(ctx context.Context, req admission.Request) ad
...
-			return admission.Errored(http.StatusBadRequest, err)
+		credentialVolumeMounts := []corev1.VolumeMount{
+			{Name: SidecarContainerWICredentialConfigMapVolumeName, MountPath: SidecarContainerWICredentialConfigMapVolumeMountPath},
+		}
+		var envVars []corev1.EnvVar
+		if credentialConfig.CredentialSource.Executable == nil {
+			mountPath := filepath.Dir(credentialConfig.CredentialSource.File)
+			credentialVolumeMounts = append(credentialVolumeMounts, corev1.VolumeMount{Name: SidecarContainerWITokenVolumeName, MountPath: mountPath})
+		} else {
+			envVars = append(envVars, corev1.EnvVar{Name: "GOOGLE_EXTERNAL_ACCOUNT_ALLOW_EXECUTABLES", Value: "1"})
```
Comment (amacaskill) [inline pkg/webhook/mutatingwebhook.go]: add constant  for this

File: pkg/webhook/mutatingwebhook.go (hunk 1 above)
Comment (amacaskill) [reply pkg/webhook/mutatingwebhook.go]: The doc you wrote says "Relative paths are rejected by the webhook and will trigger warnings or pod deployment failures." but here you have warnings

File: pkg/webhook/mutatingwebhook.go [hunk 3]
```diff
@@ -136,29 +140,78 @@ func (si *SidecarInjector) Handle(ctx context.Context, req admission.Request) ad
...
+			sidecarCredentialConfig = &SidecarContainerCredentialConfiguration{}
+		}
+		mounts := strings.Split(additionalMounts, ",")
+		for _, mount := range mounts {
+			mount = strings.TrimSpace(mount)
+			if mount == "" {
+				continue
+			}
+			parts := strings.Split(mount, ":")
+			if len(parts) == 2 {
```
Comment (amacaskill) [inline pkg/webhook/mutatingwebhook.go]: Can this be more than 2?  If there is any chance can we make this less brittle using strings.SplitN(mount, ":", 2) or strings.Cut(mount, ":") to split only on the first colon, preserving any subsequent colons in path options.

File: pkg/webhook/mutatingwebhook_test.go
Comment (amacaskill) [reply pkg/webhook/mutatingwebhook_test.go:1]: these tests were suggested by gemini:
 Incorrect Formatting Cases: Add unit tests confirming that invalid formats—such as paths containing no colons (volumePath) or too many colons (volume:path:extra)—are correctly ignored or trigger appropriate warnings without breaking the other valid mounts in the list.
Empty Slices and Boundary Delimiters: Add test cases validating how the parsing logic handles leading, trailing, or duplicate separators (e.g., ,auth-cli-vol:/etc/meta,,meta-ca-vol:/etc/meta/ca,).
Empty Keys and Values: Verify that malformed definitions like :path or volName: are safely rejected and do not lead to invalid Kubernetes API mutations.
Volume Type Independence: Ensure there is a test asserting that the webhook correctly matches and mounts a volume of the native image volume type within pod.Spec.Volumes, validating the exact mapping that Meta is deploying for Option A.

**Review by amacaskill — APPROVED** [review]

File: pkg/webhook/mutatingwebhook.go (hunk 1 above)
Comment (yangspirit) [reply pkg/webhook/mutatingwebhook.go]: Done

File: pkg/webhook/mutatingwebhook_test.go
Comment (yangspirit) [reply pkg/webhook/mutatingwebhook_test.go:1]: Done.

File: pkg/webhook/mutatingwebhook.go (hunk 3 above)
Comment (yangspirit) [reply pkg/webhook/mutatingwebhook.go]: Done

File: pkg/webhook/mutatingwebhook.go (hunk 2 above)
Comment (yangspirit) [reply pkg/webhook/mutatingwebhook.go]: Done.

**Review by amacaskill — APPROVED** [review]

File: pkg/webhook/mutatingwebhook.go (hunk 3 above)
Comment (amacaskill) [reply pkg/webhook/mutatingwebhook.go]: Do you need to push your changes? I still see 
```
		parts := strings.Split(mount, ":")
			if len(parts) == 2 {
```

Comment (amacaskill) [reply pkg/webhook/mutatingwebhook.go]: GKE internal PR looks correct, and uses strings.Cut

Comment (amacaskill) [reply pkg/webhook/mutatingwebhook.go]: NVM looked at wrong commit
